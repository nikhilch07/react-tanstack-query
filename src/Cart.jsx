import React from "react";
import { intl } from "./utils/formatCurrency";

function Cart({ cart, checkout }) {
  const total = cart.reduce((acc, item) => {
    return acc + item.pizza.sizes[item.pizzaSize];
  }, 0);
  return (
    <div className="cart">
      <h2>Cart</h2>
      <ul>
        {cart.map((item, index) => (
          <li key={index}>
            <span className="type">{item.pizza.name} - </span>
            <span className="size">{item.pizzaSize} - </span>
            <span className="size">{item.price} </span>
          </li>
        ))}
      </ul>
      <p> Total: {intl.format(total)}</p>
      <button onClick={checkout}>Checkout</button>
    </div>
  );
}

export default Cart;
