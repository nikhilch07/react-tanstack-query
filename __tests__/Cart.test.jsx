import { expect, test } from 'vitest';
import { render } from '@testing-library/react';
import Cart from '../src/Cart';

test("snapshot with nothing in cart", () => {
    const { asFragment } = render(<Cart cart={[]} />);
    expect(asFragment()).toMatchSnapshot();
});

test("snapshot with items in cart", () => {
    
    const cart = [
        {
            pizza: {
                name: "Pizza 1",
                sizes: {
                    S: 10,
                    M: 15,
                    L: 20
                }
            },
            pizzaSize: "S"
        },
        {
            pizza: {
                name: "Pizza 2",
                sizes: {
                    S: 10,
                    M: 15,
                    L: 20
                }
            },
            pizzaSize: "M"
        }
    ];
    const { asFragment } = render(<Cart cart={cart} />);
    expect(asFragment()).toMatchSnapshot();
});
