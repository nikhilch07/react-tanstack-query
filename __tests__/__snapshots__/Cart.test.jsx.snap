// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`snapshot with items in cart 1`] = `
<DocumentFragment>
  <div
    class="cart"
  >
    <h2>
      Cart
    </h2>
    <ul>
      <li>
        <span
          class="type"
        >
          Pizza 1 - 
        </span>
        <span
          class="size"
        >
          S - 
        </span>
        <span
          class="size"
        >
           
        </span>
      </li>
      <li>
        <span
          class="type"
        >
          Pizza 2 - 
        </span>
        <span
          class="size"
        >
          M - 
        </span>
        <span
          class="size"
        >
           
        </span>
      </li>
    </ul>
    <p>
       Total: $25.00
    </p>
    <button>
      Checkout
    </button>
  </div>
</DocumentFragment>
`;

exports[`snapshot with nothing in cart 1`] = `
<DocumentFragment>
  <div
    class="cart"
  >
    <h2>
      Cart
    </h2>
    <ul />
    <p>
       Total: $0.00
    </p>
    <button>
      Checkout
    </button>
  </div>
</DocumentFragment>
`;
